### PR TITLE
[Metricbeat/Kibana/stats] Enforce `exclude_usage=true`

### LINKED ISSUE #22651: [Metricbeat][Kibana][stats] Stop collecting usage
**Describe the enhancement:**

As part of the efforts for decoupling Monitoring & Telemetry in Kibana (https://github.com/elastic/kibana/issues/83626), we are stopping the collection of usage data (https://github.com/elastic/kibana/issues/83521), meaning we'll only collect _metric-related_ stats from 7.11 and future versions.

This can be achieved by always sending `exclude_usage=true` when requesting the API `/api/stats?extended=true&legacy=true&exclude_usage=true` ([line where we control it](https://github.com/elastic/beats/blob/master/metricbeat/module/kibana/stats/stats.go#L166)).

**Describe a specific use case for the enhancement or feature:**

Currently, the `stats` dataset in the Metricbeat-Kibana module collects the extended usage details every 24h so the Monitoring plugin store it in the `.monitoring-kibana-*` indices. These usage details are not actually used by the Monitoring plugin itself to anything other than reporting telemetry.

With the rise of the `local` telemetry, `monitoring`-sourced telemetry doesn't need to report this data anymore, so we can disable the collection in the 2 sources of information for the Monitoring plugin:

- Legacy Kibana-driven collection (https://github.com/elastic/kibana/pull/83546)
- Metricbeat collection (this issue).

### PR DESCRIPTION
**Pull request by afharo**
## What does this PR do?

Changes to the Metricbeat's `kibana` module: it enforces `exclude_usage=true` (only for the versions it's accepted).

The reason for this change is to disable the usage collection via the `/api/stats` endpoint, as it is no longer needed for their initial Telemetry purposes.

## Why is it important?

We are working towards decoupling Monitoring and Telemetry in Kibana. At the moment, Monitoring holds in the `.monitoring-kibana-*` indices the Kibana's Usage details only to be able to report them when sending the telemetry reports. This data is already sent to the remote telemetry cluster via Kibana's `local` collectors, so there's no need for this duplication in the efforts. Thanks to these changes, Monitoring will only maintain in its indices the data they really need for its own features.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] Not a Go developer myself. Did I mess anything up? 😇 

## How to test this PR locally

Setup Metricbeat with the `kibana-xpack` module enabled and notice it never asks for the `/api/stats?extended=true&legacy=true&exclude_usage=false` request (the last query parameter is always `true`).

## Related issues

- Closes #22651
- Superseds #22654
- Relates https://github.com/elastic/kibana/issues/83521
- Relates https://github.com/elastic/kibana/issues/83626

## Use cases

Explained in the issue #22651

### CODE REVIEW

**Comment by afharo** [comment]
jenkins, test this please

File: metricbeat/module/kibana/stats/stats.go [hunk 1]
```diff
@@ -162,26 +157,12 @@ func (m *MetricSet) fetchStats(r mb.ReporterV2, now time.Time) error {
...
-		content, err = m.statsHTTP.FetchContent()
-		if err != nil {
-			if shouldCollectUsage {
-				// When errored in collecting the usage stats it may be counterproductive to try again on the next poll, try to collect the stats again after usageCollectionBackoff
-				m.usageNextCollectOn = now.Add(usageCollectionBackoff)
-			}
-			return err
-		}
+		m.statsHTTP.SetURI(origURI + "&exclude_usage=true")
+	}
```
Comment (ycombinator) [inline metricbeat/module/kibana/stats/stats.go:161]: With this change we no longer need the comment above `if m.isUsageExcludable`.

File: metricbeat/module/kibana/stats/stats_test.go [hunk 2]
```diff
@@ -78,39 +75,25 @@ func TestFetchUsage(t *testing.T) {
...
-		},
-	}
-
-	for name, test := range cases {
-		t.Run(name, func(t *testing.T) {
-			m := MetricSet{
-				usageLastCollectedOn: test.usageLastCollectedOn,
-				usageNextCollectOn:   test.usageNextCollectOn,
-			}
+func TestFetchNoExcludeUsage(t *testing.T) {
```
Comment (ycombinator) [inline metricbeat/module/kibana/stats/stats_test.go:78]: Thank you for adding this test. 👍

**Review by ycombinator — APPROVED** [review]
Just left a small comment about an outdated comment. Otherwise LGTM. Nice work!

File: metricbeat/module/kibana/stats/stats.go (hunk 1 above)
Comment (afharo) [reply metricbeat/module/kibana/stats/stats.go:161]: Well spotted! I updated it to explain what's actually happening now :)
Thank you!

**Pull request merged**